### PR TITLE
[ALOY-1619] - WIP - added drawerlayout template app

### PR DESCRIPTION
**Pull request by Topener**
JIRA master ticket: https://jira.appcelerator.org/browse/ALOY-1618

This is still a work in progress.

This PR adds a cross platform drawerlayout template. Looking for feedback on implementation!

Todo:

- [ ] Form of navigation
- [ ] Add proper readme

### CODE REVIEW

File: templates/drawer_layout/README [hunk 1]
```diff
@@ -0,0 +1 @@
+Basic "Drawer Layout" application using the Alloy MVC framework.
```
Comment (hansemannn) [inline templates/drawer_layout/README]: We should use a proper Markdown formatted `README.md` that users can inspect. Including an intro, a screenshot/gif, example usage, bug-reporting (jira/slack/SO) and license (Apache 2)

File: templates/drawer_layout/app/controllers/Drawer.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+function toggleDrawer(e) {
+	$.drawer.toggleLeft();
```
Comment (hansemannn) [inline templates/drawer_layout/app/controllers/Drawer.js]: Can you lint with the eslint-alloy preset? @ewanharris Did we add the .eslintrc to Alloy templates already as well? Would be good to have it in all templates, incl. instructions on how to configure eslint as part of the readme.

File: templates/drawer_layout/app/lib/ui.drawer.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	
+	this.win.leftNavButton = buttonImg;
+};
+
+drawerWindow.prototype.toggleLeft = function() {
+	this.showDrawer();
+};
+
+drawerWindow.prototype.add = function(view) {
+	 if (view.role === 'centerView'){
```
Comment (hansemannn) [inline templates/drawer_layout/app/lib/ui.drawer.js]: Missing space. Should be resolved automatically once eslint is configured.

File: templates/drawer_layout/app/lib/ui.drawer.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+drawerWindow.prototype.getView = function() {
+	return this.win;
+};
+
+drawerWindow.prototype.addEventListener = function(event, cb) {
+	this.menuClickCb = cb;
+};
+
+drawerWindow.prototype.showDrawer = function() {
+	var navWin = Alloy.createController(this.navView.src,this.navView).getView();
```
Comment (hansemannn) [inline templates/drawer_layout/app/lib/ui.drawer.js]: Missing space as well. I will not comment on more style issues, as they will be resolved by eslint as well.

File: templates/drawer_layout/app/styles/app.tss [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+'Window': {
+	backgroundColor: '#fff' // white background instead of default transparent or black
+}
+
+'TextField[platform=android]': {
+	height: Ti.UI.SIZE
+}
+
+*/
```
Comment (hansemannn) [inline templates/drawer_layout/app/styles/app.tss]: Remove unused code

File: templates/drawer_layout/app/controllers/index.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+/**
+ * By using Drawer as our main controller we allow the option to insert a login
+ * screen in between the start of the app and opening the "main" page.
+ * Recommendation: if  you do add login, create a login controller, and decide
+ * in this controller if you want to open login or Drawer. 
+ */
+Alloy.createController('Drawer').getView().open();
```
Comment (hansemannn) [inline templates/drawer_layout/app/controllers/index.js]: I think we should use lowercase/camelCase-named controllers.

File: templates/drawer_layout/app/styles/app.tss (hunk 5 above)
Comment (Topener) [reply templates/drawer_layout/app/styles/app.tss]: Not unused code, but rather samples provided also in other templates

**Comment by hansemannn** [comment]
@Topener Are we ready to merge here? @ewanharris Can you also give it a quick shot?

**Comment by Topener** [comment]
@hansemannn not really done so far. I think I should close the PR until I get around to fix this.

**Comment by ewanharris** [comment]
@topener any updates on this one?

**Comment by hansemannn** [comment]
Sad to see this close. Still the old app templates from > 5 years ago while others present modern examples like drawers, maps, navigation and social.

**Pull request closed without merging**